stl/inc/yvals.h

```diff
@@ -221,21 +221,32 @@ _EMIT_STL_ERROR(STL1006, "_CONTAINER_DEBUG_LEVEL has been removed. It was supers
 #define _MSVC_STL_HARDENING_VECTOR _MSVC_STL_HARDENING
 #endif
 
-#ifndef _STL_CRT_SECURE_INVALID_PARAMETER
+#ifdef _STL_CRT_SECURE_INVALID_PARAMETER
+_EMIT_STL_ERROR(STL1007, "_STL_CRT_SECURE_INVALID_PARAMETER has been removed. "
+                         "It was superseded by _MSVC_STL_DOOM_FUNCTION.");
+#endif
+
 #ifdef _STL_CALL_ABORT_INSTEAD_OF_INVALID_PARAMETER
-#define _STL_CRT_SECURE_INVALID_PARAMETER(expr) _CSTD abort()
-#elif defined(_DEBUG) // Avoid emitting unused long strings for function names; see GH-1956.
-// static_cast<unsigned int>(__LINE__) avoids warning C4365 (signed/unsigned mismatch) with the /ZI compiler option.
-#define _STL_CRT_SECURE_INVALID_PARAMETER(expr) \
-    ::_invalid_parameter(_CRT_WIDE(#expr), L"", __FILEW__, static_cast<unsigned int>(__LINE__), 0)
-#else // ^^^ defined(_DEBUG) / !defined(_DEBUG) vvv
-#define _STL_CRT_SECURE_INVALID_PARAMETER(expr) _CRT_SECURE_INVALID_PARAMETER(expr)
-#endif // ^^^ !defined(_DEBUG) ^^^
-#endif // !defined(_STL_CRT_SECURE_INVALID_PARAMETER)
+_EMIT_STL_ERROR(STL1008, "_STL_CALL_ABORT_INSTEAD_OF_INVALID_PARAMETER has been removed. "
+                         "It was superseded by _MSVC_STL_USE_ABORT_AS_DOOM_FUNCTION.");
+#endif
+
+// The STL's "doom function" can be replaced. Notes:
+// * It must not throw. (Attempting to throw would slam into noexcept.)
+// * Common case: If it doesn't return, it should be marked as `[[noreturn]]`.
+// * Uncommon case: If it returns, the STL will attempt to "continue on error", behaving as if no checking was done.

```

Comment:
should we include a comment explaining _why_ one would replace the "doom function" (for 'continue as new'-behavior) as well as a warning that doing so as at once's own risk?

Reply:
Hmm, mentioning a possible rationale would be reasonable, although I'd like to avoid a long essay about all the scenarios in which one might want to continue-on-error.

Adding a warning that imminent UB is likely hazardous to one's health would be a great idea - I'll do that too (and I'll spend several lines on that one).